library/src/google-maps-provider.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {useState, useEffect, PropsWithChildren} from 'react';
+
+// https://developers.google.com/maps/documentation/javascript/url-params
+export interface GoogleMapsAPIUrlParameters {
+  libraries?: string[];
+  language?: string;
+  region?: string;
+  version?: string;
+  authReferrerPolicy?: string;
+}
+
+export interface GoogleMapsProviderProps extends GoogleMapsAPIUrlParameters {
+  googleMapsAPIKey: string;
+  mapContainer?: HTMLElement | null;
+  mapOptions?: google.maps.MapOptions;
+  onLoadScript?: () => void;
+  onLoadMap?: (map: google.maps.Map) => void;
+}
+
+/**
+ * The Google Maps map context
+ */
+export const GoogleMapsMapContext = React.createContext<
+  google.maps.Map | undefined
+>(undefined);
+
+/**
+ * The global Google Maps provider
+ */
+export const GoogleMapsProvider: React.FunctionComponent<
+  PropsWithChildren<GoogleMapsProviderProps>
+> = props => {
+  const {
+    children,
+    googleMapsAPIKey,
+    mapContainer,
+    mapOptions,
+    libraries,
+    language,
+    region,
+    version,
+    authReferrerPolicy,
+    onLoadScript,
+    onLoadMap
+  } = props;
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [map, setMap] = useState<google.maps.Map>();
+
+  // Load Google Maps API
+  useEffect(() => {
+    if (google?.maps) {
```

Comment:
Does this actually work? I read that the following way is actually the right way to check:
```suggestion
    if (typeof google === 'object' && typeof google.maps === 'object') {
```

Reply:
Ok, I can change this.
But what's the actual advantage here? `google.maps` could still be any type of object, so this isn't really a guarantee to be the Google Maps API. 🤔